librarian/hooks.py

```diff
@@ -0,0 +1,92 @@
+import importlib
+
+from fsal.client import FSAL
+from ondd_ipc.ipc import ONDDClient
+
+from .core.exts import ext_container as exts
+from .data.notifications import Notification
+from .data.setup import Setup, SetupWizard
+from .helpers.notifications import invalidate_notification_cache
+from .presentation.dashboard.registry import DashboardPluginRegistry
+from .presentation.menu.registry import MenuItemRegistry
+from .presentation.settings import SettingsManager
+
+
+def import_attr(path):
+    separated = path.split('.')
+    mod_path = '.'.join(separated[:-1])
+    attr_name = separated[-1]
+    mod = importlib.import_module(mod_path)
+    return getattr(mod, attr_name)
+
+
+def install_extensions(supervisor):
+    exts.fsal = FSAL(exts.config['fsal.socket'])
+    exts.dashboard = DashboardPluginRegistry()
+    exts.menuitems = MenuItemRegistry()
+    exts.notifications = Notification
+    exts.notifications.on_send(invalidate_notification_cache)
+    exts.ondd = ONDDClient(exts.config['ondd.socket'])
+    exts.settings = SettingsManager(supervisor)
+    exts.setup = Setup(supervisor)
+    exts.setup_wizard = SetupWizard(name='setup')
+
+
+def register_commands():
+    for command_path in exts.config['command.handlers']:
+        command_cls = import_attr(command_path)
+        exts.commands.register(command_cls.name,
+                               command_cls(),
+                               command_cls.option,
+                               command_cls.action,
+                               command_cls.help)
+
+
+def register_menu_items():
```

Comment:
if we unify the apis of the registration methods, that is all can just accept a class without extra calls, we can generalize these register_xx_ functions